extension/getRootCategories.js

```diff
@@ -0,0 +1,98 @@
+const fetch = require('node-fetch')
+const RootCategories = require('./models/catalog/rootCategories/rootCategories')
+let Shopify = null
+
+/**
+ * @param context
+ * @param input
+ * @param cb
+ * @returns {Promise.<void>}
+ */
+module.exports = async (context, input, cb) => {
+  Shopify = require('./lib/shopify.api')(context.config)
+
+  try {
+    /* Needs to be an Array because of the specifications */
+    const rootCategories = []
+    rootCategories.push(await getRootCategories())
+
+    cb(null, {categories: rootCategories})
+  } catch (err) {
+    cb(err)
+  }
+}
+
+/**
+ * Get Root-Categories
+ * @returns {Promise.<RootCategories>}
+ */
+async function getRootCategories () {
+  const response = await fetch(
+    Shopify.getGraphQlUrl(),
+    Shopify.getGraphQlApiRequestHeader(JSON.stringify(getGraphQlBody()))
+  )
+
+  if (!response) {
+    throw new Error('Invalid resonse.')
+  }
+
+  const json = await response.json()
+  const rootCategories = new RootCategories()
+  rootCategories.addCategories(json.data.shop.collections.edges)
+
+  //Get product count for each category
+  if (Array.isArray(rootCategories)) {
+    rootCategories.RootCategories.forEach(async (rootCategory) => {
+      rootCategory.productCount = await getProductCount(rootCategory.id)
+    })
+
+    return rootCategories
+  }
+
+  rootCategories.productCount = await getProductCount(rootCategories.id)
+  return rootCategories
+}
+
+/**
+ * Return the Query-Body which will be send to the GraphQl-API
+ * @returns {{query: string}}
+ */
+function getGraphQlBody () {
+  return {
+    query: `
+    query { 
+      shop { 
+        collections (first: 250) { 
```

Comment:
unfortunately - whats happens if the shop has more than 250 collection ?

Reply:
So, if we want to implement some kind of pagination, we have to store the "cursor" of the last collection we have and provide this cursor to the next request on this pipeline. This is not a trivial thing here. Since the Shopify-Api only supports this parameters for the collections:

```
first: Int
after: String
last: Int
before: String
reverse: Boolean = false
sortKey: CollectionSortKeys = ID
```
So, how many Shops are using more than 250 Root-Categories? We can increase the number to 500 for example, but i don't think this is necessary. 

@SG-Noxoreos What do you think about it?

Reply:
250 is the limit. some merchants have thousands of categories. Usually those are auto generated using the Admin API.